src/main/haskell/kore/src/Kore/Unification/UnifierImpl.hs

```diff
@@ -249,9 +253,8 @@ normalizeSubstitutionDuplication tools substitutionSimplifier subst =
                     varAndSubstList
             (finalSubst, proof) <-
                 normalizeSubstitutionDuplication tools substitutionSimplifier
-                    (concat singletonSubstitutions
-                     ++ Predicated.substitution predSubst
-                    )
+                    $ (Substitution.wrap $ concat singletonSubstitutions)
+                     <> Predicated.substitution predSubst
```

Comment:
Strange indentation level, I'd use 3 more spaces here (admittedly, it was somewhat strange in the original version)